system-x/services/google/pubsub/src/main/java/software/tnb/google/cloud/pubsub/validation/PubSubValidation.java

```diff
@@ -0,0 +1,134 @@
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class PubSubValidation {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PubSubValidation.class);
+
+    private TopicAdminClient topicAdminClient;
+    private SubscriptionAdminClient subscriptionAdminClient;
+    private String projectId;
+    private CredentialsProvider credsProvider;
+
+    public PubSubValidation(TopicAdminClient topicAdminClient, SubscriptionAdminClient subscriptionAdminClient, CredentialsProvider credsProvider,
+        String projectId) {
+        this.topicAdminClient = topicAdminClient;
+        this.subscriptionAdminClient = subscriptionAdminClient;
+        this.projectId = projectId;
+        this.credsProvider = credsProvider;
+    }
+
+    public Topic createNewTopic(String topicId) {
+        Topic topic = topicAdminClient.createTopic(TopicName.of(projectId, topicId));
+        LOG.debug("Created google pubsub topic: " + topic.getName());
+        return topic;
+    }
+
+    public void deleteTopic(String topicId) {
+        topicAdminClient.deleteTopic(TopicName.of(projectId, topicId));
+        LOG.debug("Deleted google pubsub topic: " + topicId);
+    }
+
+    public Subscription createNewSubscription(String subscriptionId, String topicId) {
+        SubscriptionName subscriptionName = SubscriptionName.of(projectId, subscriptionId);
+        TopicName topicName = TopicName.of(projectId, topicId);
+
+        Subscription subscription = subscriptionAdminClient.createSubscription(subscriptionName, topicName, PushConfig.getDefaultInstance(), 10);
+        LOG.debug("Created google pubsub subscription: " + subscription.getName());
+        return subscription;
+    }
+
+    public void deleteSubscription(String subscriptionId) {
+        subscriptionAdminClient.deleteSubscription(SubscriptionName.of(projectId, subscriptionId));
+        LOG.debug("Deleted google pubsub subscription: " + subscriptionId);
+    }
+
+    public String publishMessage(String topicId, String message) {
+        Publisher publisher = null;
+        String messageId;
+        try {
+            // Create a publisher instance with default settings bound to the topic
+            publisher = Publisher.newBuilder(TopicName.of(projectId, topicId)).setCredentialsProvider(credsProvider).build();
+
+            ByteString data = ByteString.copyFromUtf8(message);
+            PubsubMessage pubsubMessage = PubsubMessage.newBuilder().setData(data).build();
+
+            // Once published, returns a server-assigned message id (unique within the topic)
+            ApiFuture<String> messageIdFuture = publisher.publish(pubsubMessage);
+            messageId = messageIdFuture.get();
+            LOG.info("Published message ID: " + messageId);
```

Comment:
is the `messageId` relevant for something? if not, I would rather log that the message `x` was published to topic `y` (to be able to distinguish between the invocations when publishing multiple messages in a test?)